pkg/interactor/list_data_test.go

```diff
@@ -0,0 +1,210 @@
+package interactor
+
+import (
+	"errors"
+	"testing"
+
+	thingentities "github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/entities"
+	"github.com/CESARBR/knot-cloud-storage/pkg/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+type ListDataTestCase struct {
+	name             string
+	authorization    string
+	query            *entities.Query
+	fakeLogger       *mocks.FakeLogger
+	fakeDataStore    *mocks.FakeDataStore
+	fakeThingService *mocks.FakeThingService
+}
+
+var ldCases = []ListDataTestCase{
+	{
+		"authorization key not provided",
+		"",
+		&entities.Query{},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{},
+		&mocks.FakeThingService{
+			ReturnErr: errors.New("no access token provided"),
+		},
+	},
+	{
+		"successful get on the database",
+		"authorization-key",
+		&entities.Query{},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{
+			Data: []entities.Data{
+				{
+					From: "fc3fcf912d0c290a",
+					Payload: entities.Payload{
+						SensorID: 1,
+						Value:    3,
+					},
+					Timestamp: date,
+				},
+			},
+		},
+		&mocks.FakeThingService{
+			Thing: []*thingentities.Thing{
+				{
+					ID:    "fc3fcf912d0c290a",
+					Token: "token",
+					Name:  "thing",
+					Schema: []thingentities.Schema{
+						{
+							SensorID:  1,
+							ValueType: 3,
+							Unit:      0,
+							TypeID:    65521,
+							Name:      "Test",
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		"successful get on the database by thingID",
+		"authorization-key",
+		&entities.Query{
+			ThingID: "fc3fcf912d0c290a",
+		},
+		&mocks.FakeLogger{},
+		&mocks.FakeDataStore{
+			Data: []entities.Data{
+				{
+					From: "fc3fcf912d0c290a",
+					Payload: entities.Payload{
+						SensorID: 1,
+						Value:    3,
+					},
+					Timestamp: date,
+				},
+			},
+		},
+		&mocks.FakeThingService{
+			Thing: []*thingentities.Thing{
+				{
+					ID:    "fc3fcf912d0c290a",
+					Token: "token",
+					Name:  "thing",
+					Schema: []thingentities.Schema{
+						{
+							SensorID:  1,
+							ValueType: 3,
+							Unit:      0,
+							TypeID:    65521,
+							Name:      "Test",
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		"successful get on the database filtered by sensorID",
+		"authorization-key",
+		&entities.Query{
+			SensorID: "1",
```

Comment:
Is this passing? Because when listing by `sensorID` we should fill the query with a `deviceID` too.

Reply:
Unfortunately, yes 😢

Reply:
Could you fix it, please?

Reply:
Done!